ui-guide/Components/Modals.purs

```diff
@@ -50,9 +54,16 @@ component =
   where
     eval :: Query ~> H.ParentDSL State Query (ChildQuery m) ChildSlot Message m
     eval = case _ of
-      Open a -> a <$ H.put true
+      HandleKey ev reply -> do
+        Modal.ifCloseThen ev reply (H.put false)
```

Comment:
`Modal.whenClose`? `Modal.whenEsc`?

Reply:
It's not the same meaning as `when` though because it doesn't end in `pure unit`. It's always resulting in some `a` value, so it's close to `if ... then do ... else do ...`. I'm not against this name otherwise. @foresttoney?

Reply:
`Modal.onEscape`?